src/lib/parameters/px4params/jsonout.py

```diff
@@ -0,0 +1,119 @@
+                    curr_param['type'] = param.GetType().capitalize()
+                    if not curr_param['type'] in allowed_types:
+                        print("Error: %s type not supported: curr_param['type']" % (curr_param['name'],curr_param['type']) )
+                        sys.Exit(1)
+
+                    curr_param['group'] = group_name
+                    if (param.GetCategory()):
+                        curr_param['category'] = param.GetCategory()
+
+                    if (param.GetVolatile() == "true"):
+                        curr_param['volatile'] = param.GetVolatile()
+
+                    last_param_name = param.GetName()
+                    for code in param.GetFieldCodes():
+                        value = param.GetFieldValue(code)
+                        if code == "board":
+                            if value == board:
+                                board_specific_param_set = True
+                                # JSON schema has no field for board_specific schema. Ignore.
+                            else:
+                                #xml_group.remove(xml_param)
+                                continue
+                        else:
+                            #map PX4 param field names to schema names
+                           if code in schema_map:
+                               curr_param[schema_map[code]] = value
+                           else:
+                               print('ERROR: Field not in json schema: %s' % code)
+                               sys.exit(1)
+
+
+                if last_param_name != param.GetName():
+                    board_specific_param_set = False
+
+                enum_codes=param.GetEnumCodes() or '' # Gets numerical values for parameter.
+                if enum_codes:
+                    enum_codes=sorted(enum_codes,key=float)
+                    codes_list=list()
+                    for item in enum_codes:
+                        code_dict=dict()
+                        code_dict['value']=item
+                        code_dict['description']=param.GetEnumValue(item)
+                        codes_list.append(code_dict)
+                    curr_param['values'] = codes_list
+
+
+                if len(param.GetBitmaskList()) > 0:
+                    bitmasks_list=list()
+                    for index in param.GetBitmaskList():
+                        bitmask_dict=dict()
+                        bitmask_dict['index']=index
+                        bitmask_dict['description']=param.GetBitmaskBit(index)
+                        bitmasks_list.append(bitmask_dict)
+                    curr_param['bitmask'] = bitmasks_list
+
+
+                all_params.append(curr_param)
+
+
+        #Json string output.
+        self.output = json.dumps(all_json,indent=2)
+
+
+
+
+    def Save(self, filename):
+        with codecs.open(filename, 'w', 'utf-8') as f:
+            f.write(self.output)
+
+        #create gz version
+        gz_filename=filename+'.gz'
```

Comment:
I prefer to have the compression as a separate step, to make it more explicit, less surprising and easier to change.

Reply:
Thanks @bkueng - did that by adding new method and a --compress option (off by default). It will be ignored of course if the output does not define this method. I further added this option to cmake.